### PR TITLE
TracyExtension: added options logDirectory and logSeverity

### PR DESCRIPTION
**Pull request by JanTvrdik**
Not tested, but hopefully it will work =)

### CODE REVIEW

**Comment by fprochazka** [comment]
The logDirectory would be a bit schizophrenic now.

File: src/Bridges/Nette/TracyExtension.php [hunk 1]
```diff
@@ -19,6 +19,8 @@ class TracyExtension extends Nette\DI\CompilerExtension
 {
 	public $defaults = array(
 		'email' => NULL,
+		'logDirectory' => NULL,
+		'logSeverity' => 0,
```
Comment (dg) [inline src/Bridges/Nette/TracyExtension.php]: Why not NULL?

**Comment by JanTvrdik** [comment]
Probably a noob question, but how do you write `E_ALL` in NEON? I tried

~~~
tracy:
	logSeverity: E_ALL
~~~

and

~~~
tracy:
	logSeverity: ::constant('E_ALL')
~~~

The first generates

~~~
Tracy\Debugger::$logSeverity = 'E_ALL';
~~~

and the second on generates

~~~
Tracy\Debugger::$logSeverity = Nette\PhpGenerator\Helpers::createObject('Nette\DI\Statement', array(
	"\x00Nette\\DI\\Statement\x00entity" => '::constant',
	'arguments' => array('E_ALL'),
));
~~~

File: src/Bridges/Nette/TracyExtension.php (hunk 1 above)
Comment (JanTvrdik) [inline src/Bridges/Nette/TracyExtension.php]: I know nothing about writing extension and thought that the value should correspond to default value of `Debugger::$logSeverity`.

Comment (dg) [inline src/Bridges/Nette/TracyExtension.php]: To rewrite what user sets in boostrap.php is bad. https://github.com/JanTvrdik/nette-tracy/blob/f47bdf5dccfdc26dbb61cf7c4cf1f128ef2ffd7e/src/Bridges/Nette/TracyExtension.php#L71

**Comment by dg** [comment]
Like `::E_ALL`, but TracyExtension is not ready for this. On the other hand, it to use constant is not enough, something like `E_WARNING | E_NOTICE` should be supported to.

`createObject(...)` is bug, I'll try to fix it.

File: src/Bridges/Nette/TracyExtension.php (hunk 1 above)
Comment (JanTvrdik) [inline src/Bridges/Nette/TracyExtension.php]: What if we change `if ($value !== NULL)` to `if ($value !== $this->defaults[$key])`?

**Comment by JanTvrdik** [comment]
@fprochazka Could you elaborate on why would allowing `logSeverity` in `config.neon` cause any problems?

**Comment by fprochazka** [comment]
You should set the log directory as soon as possible - which means in bootstrap - you may have errors during compilation and you need them logged. Also, having two different directories for logging in aplication is not something I would call smart. And if you have no reason to change the directory, because you have to set it in bootstrap, and you would just copy the same value, that is only a space for errors IMHO - because you're configuring the same thing from two separate places.

**Comment by fprochazka** [comment]
If the tracy extension were first and you would set it also in `loadConfiguration` directly to `Tracy\Debugger`, expecting no errors in nette ever to occure, than it would might make sense. Now it doesn't for me.

**Comment by dg** [comment]
loadConfiguration in first extension is tooooo late.

File: src/Bridges/Nette/TracyExtension.php (hunk 1 above)
Comment (dg) [inline src/Bridges/Nette/TracyExtension.php]: And change `email` etc. to `''`. Seems like unnecessary complication.

**Comment by fprochazka** [comment]
Exactly :)

File: src/Bridges/Nette/TracyExtension.php (hunk 1 above)
Comment (JanTvrdik) [inline src/Bridges/Nette/TracyExtension.php]: Maybe for Tracy, but (generally speaking) it allows you to generate optimal code which in some cases may reduces number of included files.

**Comment by JanTvrdik** [comment]
> E_WARNING | E_NOTICE should be supported to

I think that in most situation you want to log everything (E_ALL) or nothing (0).

**Comment by dg** [comment]
Probably, but what about minor situations ;-) But it can be easily solved is extensions. 

The question is, whether to configure this option is not the same issue as logDirectory. (This also applies to half the other options.)

**Pull request merged**